### PR TITLE
⚡️ implement custom scrollable widget

### LINKED ISSUE #157: firefox and brave plugins possibly show too many entries
Sqlite based plugins like the firefox history entry plugin possibly renders a too long list blocking centerpiece responsiveness. Probably the best mitigation to this is only showing entries from thies plugins when the list of entries has been filtered to less then 10 entries.

### PR DESCRIPTION
**Pull request by friedow**
Since the iced scrollable widget renders all entries passed to it, we're experiencing degraded performance when plugins pass many entries to render. This is noticeable when using plugins like brave or firefox history with large history lists. 

This PR removes the iced scrollable widget and instead creates a list of 10 enties which behave like a scrollable. This way we really only paint the 10 entries which are in the current viewport.

This fixes #157.

### CODE REVIEW

File: client/src/component/divider.rs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+pub fn view() -> iced::Element<'static, crate::Message> {
+    iced::widget::column![iced::widget::horizontal_rule(1)]
+        .padding(iced::Padding::from([
+            1. * crate::REM,
+            0.,
+            0.5 * crate::REM,
+            0.,
+        ]))
+        // We're fixing the height here to unitfy it
```
Comment (a-kenji) [inline client/src/component/divider.rs:9]: unify

File: client/src/component/entry.rs [hunk 2]
```diff
@@ -1,15 +1,22 @@
...
+                    .width(iced::Length::Fill)
+                    .shaping(iced::widget::text::Shaping::Advanced),
+                iced::widget::text(if active { &entry.action } else { "" }).size(1. * crate::REM)
+            ]
+            .padding(0.5 * crate::REM),
+        )
+        .style(style(active)),
     )
-    .style(style(active))
+    // We're fixing the height here to unitfy it
```
Comment (a-kenji) [inline client/src/component/entry.rs:15]: unify

File: client/src/component/plugin_header.rs [hunk 3]
```diff
@@ -0,0 +1,21 @@
+pub fn view(plugin: &crate::model::Plugin) -> iced::Element<'static, crate::Message> {
+    iced::widget::row![iced::widget::text(&plugin.title)
+        .font(iced::Font {
+            family: iced::font::Family::Name("FiraCode Nerd Font"),
+            weight: iced::font::Weight::Light,
+            stretch: iced::font::Stretch::Normal,
+            monospaced: true,
+        })
+        .size(0.75 * crate::REM)]
+    // We're fixing the height here to unitfy it
```
Comment (a-kenji) [inline client/src/component/plugin_header.rs:10]: unify

**Review by a-kenji — APPROVED** [review]
LGTM, tested. Seems to work well.

There is no scroll bar anymore, I guess that is intentional?

**Comment by friedow** [comment]
Yep, scrollbar is not displayed intentionally. I think thats fine for now.

**Pull request merged**